### ISSUE TITLE
Python: Plot/Widget with Multi-Sim

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/picongpu#2716]
As discussed offline for parameter scan updates:

Our `plot_mpl` and upcoming `jupyter_widget` interfaces should support a *list of simulation paths* in their constructors in order to plot multiple simulations in one plot/widget.

One can probably implement this with a `for sim_path in ...:` so one can single arguments and lists at the same time.

Note: the data readers should *not* accept a list, this just complicates loading where one can loop externally.
Note2: maybe the data readers should be moved into a sub-directory inside `lib/python/picongpu/plugins/` as well.

Note3: for imshow / 2D plots, we need to chose / create colorbars with transparent portions to make this work (e.g. phase spaces). Various simulations will over-plot each other, which is fine, as long as we keep a deterministic order.

**Comment by codingS3b** [comment]
Just wanted to add that this might be problematic for the `png_visualizer` which also only calls imshow but I don't think that transparency will help much there...the image will just be a mess I would think.

So maybe we need some abstraction there and implement it only for those visualizers for which it makes sense? Not sure of how that would look like though...

**Comment by ax3l** [comment]
You are right, I think the interface could still be the same and implementations like the one for PNG can throw or warn if more than one is given.

**Comment by ax3l** [comment]
@codingS3b while we are on it, shall we `import future` the `super()` functionality that makes e.g. `picongpu.plugins.data.EnergyHistogramData` in its `__init__` right now Python3-only?

https://stackoverflow.com/questions/38963018/typeerror-super-takes-at-least-1-argument-0-given-error-is-specific-to-any

(see both answers)

**Comment by codingS3b** [comment]
Do you still want to support python2? If that is really necessary, I would probably rather refactor the `super` statement in way that supports both python2 and python3.

**Comment by ax3l** [comment]
From what I see, just adding the future import (maybe after a version check) should be already enough, no?
